lib/wallet/wallet.js

```diff
@@ -1839,7 +1838,7 @@ class Wallet extends EventEmitter {
     output.covenant.pushHash(nameHash);
     output.covenant.pushU32(height);
     output.covenant.pushHash(nonce);
-    reveal.addOutpoint(Outpoint.fromTX(bid, bidOuputIndex));
+    reveal.addCoin(bidCoin);
```

Comment:
These are technically not necessary because the Coin would attempted to be added to CoinView if it's already not there. But this is cleaner.

Reply:
So I think now with this logic: https://github.com/handshake-org/hsd/pull/639/commits/edc1f253b37aa228ee458ecb70d9b8d41fff39f0 using `addCoin()` is actually an optimization!

Reply:
addCoin will still stay an optimization if we use my suggestion.